server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -281,6 +281,10 @@ void checkFieldLimit(long limit, int additionalFieldsToAdd) {
         }
     }
 
+    boolean exceedsLimit(long limit, int additionalFieldsToAdd) {
```

Comment:
if we don't use it and rely on merge alone, even for the first document being indexed, let's revert changes in this class?

Reply:
got it. Leave it then.